### ISSUE TITLE
html-has-lang rule is not triggered when there is no html start tag

### ISSUE THREAD

**Issue by cstrobbe** [dequelabs/axe-core#4251]
### Product

axe Linter

### Product Version

4.8.0

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation

When there is no `lang` attribute that identifies the main language of the page, I expect the html-has-lang rule to be triggered.

#### Actual

When the page has no `html` start tag, the html-has-lang rule does not check whether the main language is identified by a different means.

#### How to Reproduce

The rule is not triggered by the following code snippet:

```
<!DOCTYPE html>
  <title>Test Page for Linters</title>
    <h1>Test Page for Linters</h1>
```

Note that this is perfectly valid HTML 5 because the start and end tags for `html`, `head` and `body` are optional. However, the code fails WCAG SC 3.1.1.

#### Additional context

See the related issue #4250.

**Comment by WilcoFiers** [comment]
Axe-linter changes how it processes the file depending on whether the root element in it is an HTML element or not. This is to avoid false positives. Template files are often embedded into other files, so not having an HTML element doesn't tell us there won't be one when the page is actually rendered.

Having a doctype there does give a pretty strong indication that there won't be an HTML element. We could consider looking for those if there is no HTML element. Is this a real-world situation you encountered, or was this something you discovered when testing how the tool works? If this would let us catch common problems I do think its worth adding, but I have a hard time imagining developers would write a doctype and then forget to put in an HTML element.

**Comment by cstrobbe** [comment]
>Is this a real-world situation you encountered, or was this something you discovered when testing how the tool works?

I was testing several linters for VS Code.
